datafusion/common/src/config.rs

```diff
@@ -2521,6 +2521,10 @@ config_namespace! {
         // The input regex for Nulls when loading CSVs.
         pub null_regex: Option<String>, default = None
         pub comment: Option<u8>, default = None
+        // Whether to allow truncated rows when parsing.
+        // By default this is set to false and will error if the CSV rows have different lengths.
+        // When set to true then it will allow records with less than the expected number of columns
+        pub truncated_rows: Option<bool>, default = None
```

Comment:
Our csv truncated_rows support, will be included in DF 51.0.0, but not for DF 50.0.0.
https://github.com/apache/datafusion/pull/17465